packages/ubuntu_desktop_installer/test/configure_secure_boot_page_test.dart

```diff
@@ -0,0 +1,136 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/annotations.dart';
+import 'package:mockito/mockito.dart';
+import 'package:provider/provider.dart';
+import 'package:ubuntu_desktop_installer/l10n.dart';
+import 'package:ubuntu_desktop_installer/pages/configure_secure_boot/configure_secure_boot_model.dart';
+import 'package:ubuntu_desktop_installer/pages/configure_secure_boot/configure_secure_boot_page.dart';
+import 'package:ubuntu_test/utils.dart';
+import 'package:ubuntu_wizard/widgets.dart';
+
+import 'configure_secure_boot_page_test.mocks.dart';
+import 'widget_tester_extensions.dart';
+
+@GenerateMocks([ConfigureSecureBootModel])
+void main() {
+  ConfigureSecureBootModel buildModel({
+    bool? isFormValid,
+    bool? isConfirmationKeyValid,
+    bool? areTextFieldsEnabled,
+    String? securityKey,
+    String? confirmKey,
+    SecureBootMode? secureBootMode,
+  }) {
+    final model = MockConfigureSecureBootModel();
+    when(model.isFormValid).thenReturn(isFormValid ?? false);
+    when(model.isConfirmationKeyValid)
+        .thenReturn(isConfirmationKeyValid ?? true);
+    when(model.areTextFieldsEnabled).thenReturn(areTextFieldsEnabled ?? true);
+    when(model.secureBootMode)
+        .thenReturn(secureBootMode ?? SecureBootMode.turnOff);
+    when(model.securityKey).thenReturn(securityKey ?? '');
+    when(model.confirmKey).thenReturn(confirmKey ?? '');
+    return model;
+  }
+
+  Widget buildPage(ConfigureSecureBootModel model) {
+    return ChangeNotifierProvider<ConfigureSecureBootModel>.value(
+      value: model,
+      child: ConfigureSecureBootPage(),
+    );
+  }
+
+  Widget buildApp(WidgetTester tester, ConfigureSecureBootModel model) {
+    tester.binding.window.devicePixelRatioTestValue = 1;
+    tester.binding.window.physicalSizeTestValue = Size(960, 680);
+    return MaterialApp(
+      localizationsDelegates: localizationsDelegates,
+      home: Wizard(
+        routes: {'/': (_) => buildPage(model)},
+        onNext: (settings) => '/',
+      ),
+    );
+  }
+
+  testWidgets('security key input', (tester) async {
+    final model = buildModel(
+      secureBootMode: SecureBootMode.turnOff,
+      securityKey: 'key',
+      confirmKey: 'key',
+      areTextFieldsEnabled: true,
+    );
+    await tester.pumpWidget(buildApp(tester, model));
+
+    final fields = find.widgetWithText(ValidatedFormField, 'key');
+    expect(fields, findsNWidgets(2));
+    expect(tester.widget<ValidatedFormField>(fields.first).enabled, isTrue);
+    expect(tester.widget<ValidatedFormField>(fields.last).enabled, isTrue);
+
+    final radios = find.byTypeOf<RadioButton<SecureBootMode>>();
+    expect(radios, findsNWidgets(2));
+    expect(tester.widget<RadioButton<SecureBootMode>>(radios.first).groupValue,
+        equals(SecureBootMode.turnOff));
+    expect(tester.widget<RadioButton<SecureBootMode>>(radios.first).groupValue,
```

Comment:
Did you mean to test `radios.last` here?

Reply:
Yes :)